src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/brgemm_cpu_blocking.cpp

```diff
@@ -83,11 +83,17 @@ bool BrgemmCPUBlocking::mark_blocking_loops(LinearIR& linear_ir,
     if (stand_alone(type))
         return res;
 
-    const auto copy_b_expr = linear_ir.get_expr_by_node(brgemm->get_brgemm_copy());
-    const ov::snippets::VectorDims full_subtensor(2, get_full_dim_value());
-    copy_b_expr->get_input_port_descriptor(0)->set_subtensor(full_subtensor);
-    copy_b_expr->get_output_port_descriptor(0)->set_subtensor(full_subtensor);
-
+    const auto copy_b_expr = repacking::get_copy_b_expr(brgemm_expr);
+    if (copy_b_expr) {
+        copy_b_expr->get_input_port_descriptor(0)->set_subtensor({get_full_dim_value(), get_full_dim_value()});
+        copy_b_expr->get_output_port_descriptor(0)->set_subtensor({get_full_dim_value(), get_full_dim_value()});
+        if (with_compensations(type)) {
+            const ov::snippets::VectorDims compensations_subtensor{1, get_full_dim_value()};
+            OPENVINO_ASSERT(brgemm_expr->get_input_count() == 3, "Brgemm must have 3 inputs in case of compensations.");
+            brgemm_expr->get_input_port_descriptor(2)->set_subtensor(compensations_subtensor);
+            copy_b_expr->get_output_port_descriptor(1)->set_subtensor(compensations_subtensor);
+        }
```

Comment:
> I mean, if with_compensations is true, then we definitely have a brgemm_copy_B

You are right, I missed this when wrote this code. I refactored it in order to avoid code duplication. Thank you!